lib/active_model/serializer/adapter.rb

```diff
@@ -75,7 +75,7 @@ def meta_key
       end
 
       def root
-        serializer.json_key
+        @options.fetch(:root, serializer.json_key)
```

Comment:
Probably marginally better to have `@options.fetch(:root) { serializer.json_key }` so that the fallback is only evaluated when the key is missing.